### PR TITLE
Changes to handle opaque pointer transition

### PR DESCRIPTION
**Pull request by mariusz-sikora-at-amd**
This patch is introducing changes to SPIRVReader functions:
 - addStoreInstRecursively
 - transValueWithOpcode<OpArrayLength>
 - transGLSLExtInst

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit fdedacd90049db0b5e5d0435684703c1781e1548
<details><summary>CTS tests (Failed: 0/179976)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 35765/63748 (56.1%)</li>
<li>Failed: 0/63748 (0.0%)</li>
<li>Not Supported: 27983/63748 (43.9%)</li><li>Warnings: 0/63748 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 27857/52478 (53.1%)</li>
<li>Failed: 0/52478 (0.0%)</li>
<li>Not Supported: 24621/52478 (46.9%)</li><li>Warnings: 0/52478 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 38010/63750 (59.6%)</li>
<li>Failed: 0/63750 (0.0%)</li>
<li>Not Supported: 25740/63750 (40.4%)</li><li>Warnings: 0/63750 (0.0%)</li>
</ul></details>


</ul></details>

**Review by piotrAMD — DISMISSED** [review]

File: llpc/translator/lib/SPIRV/SPIRVReader.cpp [hunk 1]
```diff
@@ -1678,7 +1678,9 @@ void SPIRVToLLVM::addStoreInstRecursively(SPIRVType *const spvType, Value *store
   if (isTypeWithPadRowMajorMatrix(storeType)) {
     auto storePair = createLaunderRowMajorMatrix(storeType, storePointer);
     storePointer = storePair.second;
-    storeType = storePointer->getType()->getPointerElementType();
+    storeType = storePair.first;
+    // TODO: Remove this when LLPC will switch fully to opaque pointers.
+    assert(storeType == storePointer->getType()->getNonOpaquePointerElementType());
```
Comment (Flakebi) [inline llpc/translator/lib/SPIRV/SPIRVReader.cpp:1683]: `cast<PointerType>(storePointer->getType())->isOpaqueOrPointeeTypeMatches(storeType)` should work with and without opaque pointers (here and below).

Comment (mariusz-sikora-at-amd) [reply llpc/translator/lib/SPIRV/SPIRVReader.cpp:1683]: Yes, you are right. We could use it but do we want to have any checks after the transition to opaque pointers ? When opaque pointers will be "turned on" then `isOpaqueOrPointeeTypeMatches` will return always true.

Comment (Flakebi) [reply llpc/translator/lib/SPIRV/SPIRVReader.cpp:1683]: Right, I assumed we want to be able to run with or without opaque pointers for some weeks when opaque pointers are almost working?
That makes it easier to switch between the modes, one can just add `-opaque-pointers` when running amdllpc.

Comment (mariusz-sikora-at-amd) [reply llpc/translator/lib/SPIRV/SPIRVReader.cpp:1683]: OK, this is a good point !
I will update review.

**Comment by amdvlk-admin** [comment]
### Test summary for commit ecc182ab4f9080afb3184a8dc86195509c8ae244
<details><summary>CTS tests (Failed: 0/179976)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 35765/63748 (56.1%)</li>
<li>Failed: 0/63748 (0.0%)</li>
<li>Not Supported: 27983/63748 (43.9%)</li><li>Warnings: 0/63748 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 27857/52478 (53.1%)</li>
<li>Failed: 0/52478 (0.0%)</li>
<li>Not Supported: 24621/52478 (46.9%)</li><li>Warnings: 0/52478 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 38010/63750 (59.6%)</li>
<li>Failed: 0/63750 (0.0%)</li>
<li>Not Supported: 25740/63750 (40.4%)</li><li>Warnings: 0/63750 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit bf45c8a68096ea43e56acbb52cf99f2dc421a1cd
<details><summary>CTS tests (Failed: 0/179969)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 35764/63747 (56.1%)</li>
<li>Failed: 0/63747 (0.0%)</li>
<li>Not Supported: 27983/63747 (43.9%)</li><li>Warnings: 0/63747 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 27854/52475 (53.1%)</li>
<li>Failed: 0/52475 (0.0%)</li>
<li>Not Supported: 24621/52475 (46.9%)</li><li>Warnings: 0/52475 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 38007/63747 (59.6%)</li>
<li>Failed: 0/63747 (0.0%)</li>
<li>Not Supported: 25740/63747 (40.4%)</li><li>Warnings: 0/63747 (0.0%)</li>
</ul></details>


</ul></details>

**Review by Flakebi — APPROVED** [review]

**Review by amdrexu — APPROVED** [review]

**Pull request merged**